recipes/ckanweb-setup.rb

```diff
@@ -43,6 +43,15 @@
 # So we do both.
 execute "pip install supervisor"
 
+# Managed processes sometimes don't shut down properly on daemon stop,
+# leaving them 'orphaned' and resulting in duplicates.
+# Work around by issuing a stop command to the children first.
+execute "Stop children on supervisord stop" do
+	command <<-'SED'.strip + " /etc/init.d/supervisord"
+		sed -i 's/^\(\s*\)\(killproc\)/\1timeout 10s supervisorctl stop all || echo "WARNING: Unable to stop managed process(es) - check for orphans"; \2/'
```

Comment:
so killproc is not replaced?
end result is
timeout 10s supervisorctl stop all || echo "WARNING: Unable to stop managed process(es) - check for orphans"; killproc rest of command?

Reply:
That's correct. killproc is meant to kill the manager, which we still want to do, along with the children.

Most of the time, "stop all" will succeed and simply log stop messages for each stopped process. If it doesn't, though, then you get the warning and killproc gives it a second shot.